File: frontend/src/questionnaires/dto/QuestionnaireSubmissionResponseEditorDto.ts
import {SubmittedQuestionResponseDto} from "./SubmittedQuestionResponseDto.ts";
import {QuestionnaireStatus} from "./QuestionnaireStatus.ts";

export interface QuestionnaireSubmissionResponseEditorDto {
  id: number;
  name: string;
  description: string;
  questions: SubmittedQuestionResponseDto[];
  receivedPoints: number;
  maxPoints: number;
  createdAt: string;
  status: QuestionnaireStatus;
}

File: backend/src/main/java/net/dnadas/training_portal/dto/group/project/questionnaire/QuestionnaireSubmissionResponseEditorDto.java
package net.dnadas.training_portal.dto.group.project.questionnaire;

import net.dnadas.training_portal.model.group.project.questionnaire.QuestionnaireStatus;

import java.util.List;

public record QuestionnaireSubmissionResponseEditorDto(
  Long id, String name, String description, List<SubmittedQuestionResponseDto> questions,
  Integer receivedPoints, Integer maxPoints, String createdAt, QuestionnaireStatus status) {
}